__tests__/app.test.js

```diff
@@ -52,6 +52,81 @@ describe('/api/articles', () => {
             })
         })
     })
+    test('200: can sort articles by any valid column', () => {
+        return request(app).get('/api/articles?sort_by=author').expect(200).then(({body}) => {
+            expect(body.articles).toBeSortedBy('author', {descending: true})
+        })
+    })
+    test('200: order can be set to asc or desc, default desc', () => {
+        return request(app).get('/api/articles?order=asc').expect(200).then(({body}) => {
+            expect(body.articles).toBeSortedBy('created_at')
+        })
+    })
+    test('200: can be sorted by specific topic', () => {
+        return request(app).get('/api/articles?topic=mitch').expect(200).then(({body}) => {
+            expect(body.articles).toBeInstanceOf(Array)
+            expect(body.articles).toHaveLength(11)
+            body.articles.forEach(article => {
+                expect(article).toEqual(expect.objectContaining({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: 'mitch',
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    comment_count: expect.any(Number)
+                }))
+            })
+        })
+    })
+    test('404: invalid sort_by column', () => {
+        return request(app).get('/api/articles?sort_by=authorsName').expect(404).then(({body}) => {
+            expect(body.msg).toBe('property doesnt exist')
+        })
+    })
+    test('400: order isnt asc or desc', () => {
+        return request(app).get('/api/articles?order=hightolow').expect(400).then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    test('404: topic not in table', () => {
+        return request(app).get('/api/articles?topic=cheese').expect(404).then(({body}) => {
+            expect(body.msg).toBe('property doesnt exist')
+        })
+    })
+    test('400: sort_by blank', () => {
+        return request(app).get('/api/articles?sort_by=').expect(400).then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    test('400: order blank', () => {
+        return request(app).get('/api/articles?order=').expect(400).then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    test('400: topic blank', () => {
+        return request(app).get('/api/articles?topic=').expect(400).then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
```

Comment:
I think this is overkill. Usually if we supply a query with no value, the server would just ignore that. So if we give a query key with no value, i would just do the default behaviour as if we hadn't even wrote the query key